src/nvim/memline.c

```diff
@@ -2754,17 +2751,15 @@ static void ml_flush_line(buf_T *buf)
         /* The else case is already covered by the insert and delete */
         ml_updatechunk(buf, lnum, (long)extra, ML_CHNK_UPDLINE);
       } else {
-        /*
-         * Cannot do it in one data block: Delete and append.
-         * Append first, because ml_delete_int() cannot delete the
-         * last line in a buffer, which causes trouble for a buffer
-         * that has only one line.
-         * Don't forget to copy the mark!
-         */
-        /* How about handling errors??? */
-        (void)ml_append_int(buf, lnum, new_line, new_len, FALSE,
-            (dp->db_index[idx] & DB_MARKED));
-        (void)ml_delete_int(buf, lnum, FALSE);
+        // Cannot do it in one data block: Delete and append.
+        // Append first, because ml_delete_int() cannot delete the
+        // last line in a buffer, which causes trouble for a buffer
+        // that has only one line.
+        // Don't forget to copy the mark!
+        // How about handling errors???
```

Comment:
fwiw, when a comment describes a "block of logic", it makes sense to keep the pre/post lines, e.g.:

```
//
//  ...
//
```